Broken extension with sphinx 7.2.2

This extension is broken with sphinx 7.2.2. When I try to build [sssd.io](https://github.com/SSSD/sssd.io) it fails with `TypeError: unsupported operand type(s) for +: 'PosixPath' and 'str'`. It fails in both our [CI](https://github.com/SSSD/sssd.io/actions/runs/5944033770/job/16120455133?pr=73) and a local container environment. It was building correctly three weeks ago with sphinx 7.1.1. 

Similar to https://github.com/sphinx-doc/sphinx/issues/11629, but I'm not completely sure which of the two projects is at fault, so I prefer to open this ticket so that this extension's community is aware of the problem.

Full log:
```
Traceback (most recent call last):
  File "/sssd.io/.venv/lib/python3.10/site-packages/sphinx/events.py", line 96, in emit
    results.append(listener.handler(self.app, *args))
  File "/sssd.io/.venv/lib/python3.10/site-packages/sphinx_gitstamp/__init__.py", line 32, in page_context_handler
    docsrc = app.confdir + "/"
TypeError: unsupported operand type(s) for +: 'PosixPath' and 'str'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/sssd.io/.venv/lib/python3.10/site-packages/sphinx/cmd/build.py", line 298, in build_main
    app.build(args.force_all, args.filenames)
  File "/sssd.io/.venv/lib/python3.10/site-packages/sphinx/application.py", line 355, in build
    self.builder.build_update()
  File "/sssd.io/.venv/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 293, in build_update
    self.build(to_build,
  File "/sssd.io/.venv/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 363, in build
    self.write(docnames, list(updated_docnames), method)
  File "/sssd.io/.venv/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 571, in write
    self._write_serial(sorted(docnames))
  File "/sssd.io/.venv/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 581, in _write_serial
    self.write_doc(docname, doctree)
  File "/sssd.io/.venv/lib/python3.10/site-packages/sphinx/builders/html/__init__.py", line 655, in write_doc
    self.handle_page(docname, ctx, event_arg=doctree)
  File "/sssd.io/.venv/lib/python3.10/site-packages/sphinx/builders/html/__init__.py", line 1101, in handle_page
    newtmpl = self.app.emit_firstresult('html-page-context', pagename,
  File "/sssd.io/.venv/lib/python3.10/site-packages/sphinx/application.py", line 492, in emit_firstresult
    return self.events.emit_firstresult(event, *args,
  File "/sssd.io/.venv/lib/python3.10/site-packages/sphinx/events.py", line 117, in emit_firstresult
    for result in self.emit(name, *args, allowed_exceptions=allowed_exceptions):
  File "/sssd.io/.venv/lib/python3.10/site-packages/sphinx/events.py", line 107, in emit
    raise ExtensionError(__("Handler %r for event %r threw an exception") %
sphinx.errors.ExtensionError: Handler <function page_context_handler at 0x7efc876f3400> for event 'html-page-context' threw an exception (exception: unsupported operand type(s) for +: 'PosixPath' and 'str')

Extension error (sphinx_gitstamp):
Handler <function page_context_handler at 0x7efc876f3400> for event 'html-page-context' threw an exception (exception: unsupported operand type(s) for +: 'PosixPath' and 'str')
make: *** [Makefile:12: html] Error 2
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,8 @@ Changelog
 
 *Release date: TBD*
 
+* Fix Path use for Sphinx 8
+  [#16](https://github.com/jdillard/sphinx-gitstamp/pull/16)
 * Clean up how package versions are handled
 * Install pre-commit with isort, black, and flake8
   [#7](https://github.com/jdillard/sphinx-gitstamp/pull/7)

--- a/sphinx_gitstamp/__init__.py
+++ b/sphinx_gitstamp/__init__.py
@@ -32,7 +32,7 @@ def page_context_handler(app, pagename, templatename, context, doctree):
         # We have already errored about this
         pass
 
-    fullpagename = Path(app.confdir) / pagename
+    fullpagename = Path(app.confdir, pagename)
 
     # Don't barf on "genindex", "search", etc
     if not os.path.isfile("%s.rst" % fullpagename):

--- a/sphinx_gitstamp/__init__.py
+++ b/sphinx_gitstamp/__init__.py
@@ -10,8 +10,8 @@
 
 import datetime
 import os
-from pathlib import Path
 import sys
+from pathlib import Path
 
 from sphinx import errors
 

--- a/sphinx_gitstamp/__init__.py
+++ b/sphinx_gitstamp/__init__.py
@@ -10,6 +10,7 @@
 
 import datetime
 import os
+from pathlib import Path
 import sys
 
 from sphinx import errors
@@ -30,14 +31,8 @@ def page_context_handler(app, pagename, templatename, context, doctree):
     if g is None:
         # We have already errored about this
         pass
-    fullpagename = pagename
-    docsrc = ""
-    try:
-        docsrc = app.confdir + "/"
-        if docsrc != "/":
-            fullpagename = docsrc + pagename
-    except KeyError:
-        pass
+
+    fullpagename = Path(app.confdir) / pagename
 
     # Don't barf on "genindex", "search", etc
     if not os.path.isfile("%s.rst" % fullpagename):

